src/Sarif/ExtensionMethods.cs

```diff
@@ -15,6 +15,37 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public static class ExtensionMethods
     {
+        /// <summary>
+        /// Attempt to reconstruct a URI, if appropriate, using Run instance 
+        /// originalUriBaseId and uriBaseId properties. If this method cannot
+        /// successfully reconstitute an absolute URI, it will return false
+        /// and populate 'resolvedUri' with the input value.
+        /// </summary>
+        /// <param name="run"></param>
+        /// <param name="uri"></param>
+        /// <param name="resolvedUri"></param>
+        /// <returns></returns>
+        public static bool TryResolveUri(this Run run, FileLocation fileLocation, out Uri resolvedUri)
+        {
+            resolvedUri = fileLocation.Uri;
+
+            // We can't restore any absolute URIs unless someone has
+            // deconstructed them using uriBaseId + originalUriBaseIds
+            if (run.OriginalUriBaseIds == null) { return false; }
```

Comment:
>OriginalUriBaseIds [](start = 20, length = 18)

But... I think this is a bug. If `fileLocation.Uri.IsAbsoluteUri`, then you don't need any help from URI base ids, original or otherwise. So:
```
resolvedUri = fileLocation.Uri;

if (fileLocation.Uri.IsAbsoluteUri) { return true; }

if (run.OriginalUriBaseIds == null || string.IsNullOrEmpty(fileLocation.UriBaseId)) { return false; }

...
```

Reply:
yes, almost! you also need to reset resolvedUri to null in cases where you are returning false.

---
In reply to: [205979291](https://github.com/Microsoft/sarif-sdk/pull/974#discussion_r205979291) [](ancestors = 205979291)